Deprecation warning on keywords mem of LBFGS operators

abelsiqueira opened the pull request.

dpo commented:
NB: there's https://docs.julialang.org/en/v1/base/base/#Base.@deprecate and even https://docs.julialang.org/en/v1/stdlib/Test/#Test.@test_deprecated.

abelsiqueira commented:
Thanks for the links. Updated using `@deprecate` and `@test_deprecated`. Also included LSR1 and moved the definitions of deprecated methods to `deprecated.jl` and the tests to `test_deprecated.jl`.

amontoison commented:
I updated the tests in Krylov.jl, some of them used the old version `InverseLBFGSOperator(n, mem)` (CGLS and CRLS). But the project.toml should be updated to take into account the new release `1.1` of `LinearOperators.jl`. I will prepare the release 5.0 of Krylov.jl.
